target/linux/realtek/image/rtl930x.mk

```
@@ -51,6 +51,16 @@ define Device/xikestor_sks8300-8x
 endef
 TARGET_DEVICES += xikestor_sks8300-8x
 
+define Device/xikestor_sks8310-8x
+  SOC := rtl9303
+  UIMAGE_MAGIC := 0x93000000  
+  DEVICE_VENDOR := xikestor
+  DEVICE_MODEL := SKS8310-8X
+  IMAGE_SIZE := 20480k
+  $(Device/kernel-lzma)
```

Maybe the following changes will solve the boot issue without changing the `bootcmd` variable.

```suggestion
  $(Device/kernel-lzma)
  IMAGE/sysupgrade.bin := \
    pad-extra 16 | \
    append-kernel | \
    pad-to 64k | \
    append-rootfs | \
    pad-rootfs | \
    check-size | \
    append-metadata
```

---

This needs to be checked on the device. Is that just some offset/padding applied to make U-boot happy or what's that based on?

---

The stock firmware may be flashed without removing the header and the bootloaer may loads from +0x10.

```
$ hexdump -v -n 128 -C "V1.04.B01 Mar_10_2025.bin"
00000000  00 00 00 02 00 00 00 01  00 00 00 01 ce 65 af 08  |.............e..|
00000010  83 80 00 00 86 16 ba a9  67 ce c6 1e 00 b3 05 d1  |........g.......|
00000020  80 00 00 00 80 38 74 00  81 01 2b d3 05 05 02 03  |.....8t...+.....|
00000030  52 54 4b 5f 53 44 4b 00  00 00 00 00 00 00 00 00  |RTK_SDK.........|
00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000050  5d 00 00 00 04 ff ff ff  ff ff ff ff ff 00 00 6f  |]..............o|
00000060  fd ff ff a3 b7 7f 63 c5  55 81 bd 5b 5e 9d 54 f7  |......c.U..[^.T.|
00000070  9d ec 67 91 22 8e 1d 3e  86 b6 7e a6 1c 5a 08 eb  |..g."..>..~..Z..|
00000080
```